pkg/controller/nodeimage/nodeimage_controller.go

```diff
@@ -193,6 +197,16 @@ func (r *ReconcileNodeImage) Reconcile(_ context.Context, request reconcile.Requ
 
 	// If Node exists, we should create a NodeImage
 	if nodeImage == nil {
+		_, condition := nodeutil.GetNodeCondition(&node.Status, v1.NodeReady)
+		if condition.Status != v1.ConditionTrue {
+			klog.V(4).Infof("Skip to create NodeImage %s for Node has not ready yet.", request.Name)
+			return reconcile.Result{}, nil
```

Comment:
is it better to check the node ready condition in node event handler?

Reply:
Yes, I have added the check the ready and delay time in event handler.